### PR TITLE
Add markdown readme for System.Configuration.ConfigurationManager

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Issue: #59630

### CODE REVIEW

File: src/libraries/System.Configuration.ConfigurationManager/src/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+## About
+
+Provides types that support using XML configuration files (`app.config`). This package is mostly useful to assist in porting of the existing .NET Framework code that depends on XML configuration. For a modern configuration API, see [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/).
```
Comment (danmoseley) [inline src/libraries/System.Configuration.ConfigurationManager/src/README.md:3]: Should we make the deprecation statement stronger? This package only exists to support migration. There is no reason to write new code against this API unless it is in maintenance of code migrated from .NET Framework that already uses it.

Comment (MSDN-WhiteKnight) [reply src/libraries/System.Configuration.ConfigurationManager/src/README.md]: Updated

File: src/libraries/System.Configuration.ConfigurationManager/src/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+## About
+
+Provides types that support using XML configuration files (`app.config`). Only use this package when migrating the existing .NET Framework code that already uses XML configuration. When writing a new code, use [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) instead.
```
Comment (danmoseley) [inline src/libraries/System.Configuration.ConfigurationManager/src/README.md:3]:
```suggestion
Provides types that support using XML configuration files (`app.config`). This package exists only to support migrating existing .NET Framework code that already uses System.Configuration. When writing new code, use another configuration system instead, such as [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/).
```

File: src/libraries/System.Configuration.ConfigurationManager/src/System.Configuration.ConfigurationManager.csproj [hunk 3]
```diff
@@ -7,11 +7,8 @@
     <IsTrimmable>false</IsTrimmable>
     <NoWarn>$(NoWarn);CA1847</NoWarn>
     <IsPackable>true</IsPackable>
-    <PackageDescription>Provides types that support using configuration files.
-
-Commonly Used Types:
-System.Configuration.Configuration
-System.Configuration.ConfigurationManager</PackageDescription>
+    <PackageDescription>Provides types that support using XML configuration files (app.config). Only use this package when migrating the existing .NET Framework code that already uses XML configuration. When writing a new code, use Microsoft.Extensions.Configuration instead.</PackageDescription>
```
Comment (danmoseley) [inline src/libraries/System.Configuration.ConfigurationManager/src/System.Configuration.ConfigurationManager.csproj:10]:
```suggestion
    <PackageDescription>Provides types that support using XML configuration files (app.config). This package exists only to support migrating existing .NET Framework code that already uses System.Configuration. When writing new code, use another configuration system instead, such as Microsoft.Extensions.Configuration. </PackageDescription>
```

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Configuration.ConfigurationManager/src/System.Configuration.ConfigurationManager.csproj [hunk 4]
```diff
@@ -7,11 +7,8 @@
     <IsTrimmable>false</IsTrimmable>
     <NoWarn>$(NoWarn);CA1847</NoWarn>
     <IsPackable>true</IsPackable>
-    <PackageDescription>Provides types that support using configuration files.
-
-Commonly Used Types:
-System.Configuration.Configuration
-System.Configuration.ConfigurationManager</PackageDescription>
+    <PackageDescription>Provides types that support using XML configuration files (app.config). This package exists only to support migrating existing .NET Framework code that already uses System.Configuration. When writing new code, use another configuration system instead, such as Microsoft.Extensions.Configuration. </PackageDescription>
```
Comment (danmoseley) [inline src/libraries/System.Configuration.ConfigurationManager/src/System.Configuration.ConfigurationManager.csproj:10]:
```suggestion
    <PackageDescription>Provides types that support using XML configuration files (app.config). This package exists only to support migrating existing .NET Framework code that already uses System.Configuration. When writing new code, use another configuration system instead, such as Microsoft.Extensions.Configuration.</ PackageDescription>
```

**Review by danmoseley — APPROVED** [review]

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Comment by buyaa-n** [comment]
CI [test failures unrelated](https://github.com/dotnet/runtime/issues/74370)

**Pull request merged**